internal/terraform/node_resource_abstract.go

```diff
@@ -68,6 +68,9 @@ type NodeAbstractResource struct {
 
 	// The address of the provider this resource will use
 	ResolvedProvider addrs.AbsProviderConfig
+	// storedProviderConfig is the provider address retrieved from the
+	// state.
+	storedProviderConfig addrs.AbsProviderConfig
```

Comment:
I was confused about this new field never being written to in this diff (outside of tests), because it only happens in the abstract resource instance's `AttachResourceState` method. I understand why we want the field to be here, but maybe it's worth commenting that we depend on types which embed this struct to populate it, possibly specifically naming `AttachResourceState`.

Reply:
Thanks, that's a good point! (I mean that's why the field was in the outer struct in the first place)